src/Microsoft.AspNet.Mvc.Core/ControllerActionDescriptorBuilder.cs

```diff
@@ -80,6 +80,12 @@ public static IList<ControllerActionDescriptor> Build(ApplicationModel applicati
                             // for this action.
                             ReplaceAttributeRouteTokens(actionDescriptor, routeTemplateErrors);
 
+                            var attributeRouteInfo = actionDescriptor.AttributeRouteInfo;
+                            if (attributeRouteInfo.Name != null)
+                            {
+                                attributeRouteInfo.Name = attributeRouteInfo.Name + "_" + action.ActionName;
```

Comment:
I think there's some confusion here and a wire got crossed. I was expecting this fix to run the token replacement code on the route name. I reread your email and I see what you're saying there, I must have missed that you were proposing something different.

So `Products_[action]` -> `Products_Edit`.

The problems with what you're doing here are twofold:
1). This doesn't account for the use of `[controller]` or `[area]` - both of which are legal
2). This is magic - you're taking something the user wrote, and turning it into something she didn't write.